core/commonMain/src/ArgType.kt

```diff
@@ -70,17 +70,43 @@ abstract class ArgType<T : Any>(val hasParameter: kotlin.Boolean) {
                     ?: throw ParsingException("Option $name is expected to be double number. $value is provided.")
     }
 
+    companion object {
+        /**
+         * Helper for arguments that have limited set of possible values represented as enumeration constants.
+         */
+        inline fun <reified T: Enum<T>> Choice(
+            noinline toVariant: (kotlin.String) -> T = { enumValueOf(it.toUpperCase()) },
+            noinline toString: (T) -> kotlin.String = { it.toString().toLowerCase() }): Choice<T> {
+            return Choice(enumValues(), toVariant, toString)
+        }
+    }
+
     /**
      * Type for arguments that have limited set of possible values.
      */
-    class Choice(val values: List<kotlin.String>) : ArgType<kotlin.String>(true) {
+    class Choice<T: Enum<T>>(choices: Array<T>,
```

Comment:
- `Enum<T>` upper bound constraint seems to be not needed.
- it's more idiomatic to pass list of variants as `List<T>`

Reply:
> Enum<T> upper bound constraint seems to be not needed.

Reply:
It can work on list of any element type, no just enum. Why do we need enum constraint then?